vars/parseStageInfo.groovy

```diff
@@ -225,140 +189,71 @@ Map call(Map config = [:]) {
+                kwargs['default_tags'] = 'always_passes'
+            } else {
+                kwargs['default_tags'] = 'pr'
             }
-    } else if (!tag) {
-            if (env.BRANCH_NAME =~ branchTypeRE('testing')) {
-                tag = 'always_passes'
-      } else {
-                // Tags defined by commit pragmas have priority in user PRs
-                tag = get_commit_pragma_tags(result['pragma_suffix'])
-                if (!tag) {
-                    // Followed by stage defined tags
-                    tag = config['test_tag']
-          /* groovylint-disable-next-line CouldBeElvis */
-                    if (!tag) {
-                        // Otherwise use the default PR tag
-                        tag = 'pr'
-                    }
-                }
-            }
-        }
-        if (tag) {
-            tag = tag.trim()
-        }
-
-        // Get the ftest --nvme argument from either the build parameters or commit pragmas
-        ftest_arg_nvme = params.TestNvme ?: cachedCommitPragma(
-      'Test-nvme' + result['pragma_suffix'], cachedCommitPragma('Test-nvme', ftest_arg_nvme))
-
-        // Get the ftest --repeat argument from either the build parameters or commit pragmas
-        ftest_arg_repeat = params.TestRepeat ?: cachedCommitPragma(
-      'Test-repeat' + result['pragma_suffix'], cachedCommitPragma('Test-repeat', null))
-
-        // Get the ftest --provider argument from either the build parameters or commit pragmas if not
-        // already defined by the stage
-        ftest_arg_provider = ftest_arg_provider ?: params.TestProvider ?:
-            cachedCommitPragma(('Test-provider' + result['pragma_suffix']),
-            cachedCommitPragma('Test-provider', null))
-
-        // Assemble the ftest args
-        result['ftest_arg'] = ''
-        if (ftest_arg_nvme) {
-            result['ftest_arg'] += ' --nvme=' + ftest_arg_nvme
-        }
-        if (ftest_arg_repeat) {
-            result['ftest_arg'] += ' --repeat=' + ftest_arg_repeat
-        }
-        if (ftest_arg_provider) {
-            result['ftest_arg'] += ' --provider=' +  ftest_arg_provider
-        }
-        if (result['ftest_tag']) {
-            result['ftest_tag'] = result['ftest_tag'].trim()
         }
+        result['test_tag'] = getFunctionalTags(kwargs)
 
-        // Assemble the stage test tags - add the cluster size to each tag group
-        result['test_tag'] = ''
-        for (group in tag.split(' ')) {
-            result['test_tag'] += group + ',' + cluster_size + ' '
-        }
-        result['test_tag'] = result['test_tag'].trim()
+        // Get the ftest arguments
+        kwargs['default_nvme'] = ftest_arg_nvme
+        kwargs['provider'] = ftest_arg_provider
+        functional_args = getFunctionalArgs(kwargs)
+        result['ftest_arg'] = functional_args.get('ftest_arg', '')
+        result['stage_rpms'] = functional_args.get('stage_rpms', '')
```

Comment:
What's the difference between `kwargs['default_nvme'] = ftest_arg_nvme` and the nvme from  `getFunctionalArgs`?

Reply:
The `kwargs['default_nvme']` is the `default_nvme` argument passed to `vars/getFunctionalNvme.groovy`.  The `default_nvme` value is used in the return value by `vars/getFunctionalNvme.groovy` when neither the `TestNvme` build parameter nor a `Test-nvme*` commit pragma is set.  Assuming neither build or commit pragma is set, `vars/getFunctionalNvme.groovy` will return `''` for VM stages (because `default_nvme=''`) or `--nvme=auto`, `--nvme=auto_md_on_ssd`, or `--nvme=auto:-3DNAND` for the HW stages (based on the output from `get_default_nvme()`).